### PR TITLE
Fixed an issue in QuorumReader when quorum could not be selected even though 1 secondary and Primary are reachable and in sync

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description

When `QuorumReader` (used for consistency levels `Bounded Staleness` and `Strong`) is not able to select a quorum because it only retrieved a response from a single Secondary. The `ReadPrimaryAsync` call will fail because the replica set size is larger than `ReadQuorum`. Any retry there-after should include the Primary - so, that Primary and one Secondary can build quorum - otherwise as long as only one Secondary is actually reachable we can't serve the read operations even when 1 Secondary and 1 Primary should be able to build quorum.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by NaluTripician — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java [hunk 1]
```diff
@@ -303,6 +304,78 @@ public void readManyWithTimeout() throws Exception {
...
+            new FaultInjectionEndpointBuilder(FeedRange.forFullRange()) //feedRanges.get(feedRanges.size() - 1)
+                .replicaCount(2)
+                .includePrimary(false)
+                .build()
+        );
+        FaultInjectionCondition faultInjectionCondition = conditionBuilder.build();
+        FaultInjectionServerErrorResult connectTimeoutResult = FaultInjectionResultBuilders
+            .getResultBuilder(FaultInjectionServerErrorType.GONE)
+            .times(Integer.MAX_VALUE - 1)
+            .delay(Duration.ofDays(1))
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java:350]: the delay here will not play a role except for response_delay or connection_delay, so the delay actually comes from the 30s SDK retry. 

Is the intention here to return the response after the delay? asking to see whether we need to add the delay support for other exception in fault injection

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java [hunk 2]
```diff
@@ -303,6 +304,78 @@ public void readManyWithTimeout() throws Exception {
...
+    @Test(groups = { "fast" }, timeOut = 100 * TIMEOUT)
+    public void readManyWithTwoSecondariesNotReachable() throws Exception {
+        if (client.asyncClient().getConnectionPolicy().getConnectionMode() != ConnectionMode.DIRECT) {
+            throw new SkipException("Fault injection only targeting direct mode");
+        }
+
+        if (ImplementationBridgeHelpers
+            .CosmosAsyncClientHelper
+            .getCosmosAsyncClientAccessor()
+            .getEffectiveConsistencyLevel(client.asyncClient(), OperationType.Query, null) != ConsistencyLevel.BOUNDED_STALENESS) {
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java:316]: maybe also include strong consistency?

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks for the quick fix :)

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java:350]: I initially tried simulating not being able to connect to a node with high connect timeout. But we internally cap the connect timeout It allows simulating a single ConnectTimeoutException but afterwards channel gets established. So, couldn't use that one. Using an artificial delay on response here to hit the right code path in Quorum Reader

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java:316]: Done

**Pull request merged**